Common/include/grid_movement_structure.hpp

```diff
@@ -81,6 +81,191 @@ class CGridMovement {
+   */
+  CFreeFormBlending();
+
+  /*!
+   * \brief Destructor of the class.
+   */
+  virtual ~CFreeFormBlending();
+
+  /*!
+   * \brief A pure virtual member.
+   * \param[in] val_i - index of the basis function.
+   * \param[in] val_t - Point at which we want to evaluate the i-th basis.
+   */
+  virtual su2double GetBasis(short val_i, su2double val_t);
+
+  /*!
+   * \brief A pure virtual member.
+   * \param[in] val_i - index of the basis function.
+   * \param[in] val_t - Point at which we want to evaluate the derivative of the i-th basis.
+   * \param[in] val_order - Order of the derivative.
+   */
+  virtual su2double GetDerivative(short val_i, su2double val_t, short val_order);
+
+  /*!
+   * \brief A pure virtual member.
+   * \param[in] val_order - The new order of the function.
+   * \param[in] n_controlpoints - the new number of control points.
+   */
+  virtual void SetOrder(short val_order, short n_controlpoints);
+
+  /*!
+   * \brief Returns the current order of the function.
+   */
+  su2double GetOrder();
+
+  /*!
+   * \brief Returns the current degree of the function.
+   */
+  su2double GetDegree();
+};
+
+/*!
+ * \class CBSplineBlending
+ * \brief Class that defines the blending using uniform BSplines.
+ * \author T. Albring
+ * \version 4.3.0 "Cardinal"
+ */
+class CBSplineBlending : public CFreeFormBlending{
+
+private:
+  vector<su2double>          U;  /*!< \brief The knot vector for uniform BSplines on the interval [0,1]. */
+  vector<vector<su2double> > N;  /*!< \brief The temporary matrix holding the j+p basis functions up to order p. */
+  unsigned short KnotSize;       /*!< \brief The size of the knot vector. */
+
+public:
+
+  /*!
+   * \brief Constructor of the class.
+   */
+  CBSplineBlending(short val_order, short n_controlpoints);
+
+  /*!
+   * \brief Destructor of the class.
+   */
+  ~CBSplineBlending();
+
+  /*!
+   * \brief Returns the value of the i-th basis function and stores the values of the i+p basis functions in the matrix N.
+   * \param[in] val_i - index of the basis function.
+   * \param[in] val_t - Point at which we want to evaluate the i-th basis.
+   */
+  su2double GetBasis(short val_i, su2double val_t);
+
+  /*!
+   * \brief Returns the value of the derivative of the i-th basis function.
+   * \param[in] val_i - index of the basis function.
+   * \param[in] val_t - Point at which we want to evaluate the derivative of the i-th basis.
+   * \param[in] val_order - Order of the derivative.
+   */
+  su2double GetDerivative(short val_i, su2double val_t, short val_order_der);
+
+
+  /*!
+   * \brief Set the order and number of control points.
+   * \param[in] val_order - The new order of the function.
+   * \param[in] n_controlpoints - the new number of control points.
+   */
+  void SetOrder(short val_order, short n_controlpoints);
+
+};
+
+/*!
+ * \class CBezierBlending
+ * \brief Class that defines the blending using Bernteinpolynomials (Bezier Curves).
```

Comment:
Typo here (Bernstein), but please also check for extra spaces and extra new lines if you have a minute.